### PR TITLE
feat: snippets discoverablity

### LINKED ISSUE #7601: [Enhancement]Documentation and snippets enhancements

### PR DESCRIPTION
**Pull request by arunvjn**
## Description
This PR is an attempt to improve snippets' discoverability. It includes the following changes.

- Snippet icon in the property pane.
- Snippets button in API pane and the query editors.
- New icon for snippets.
- Snippet from property pane and editors should apply relevant filters.
- Changed hover state of slash commands.

Fixes #7601 

## Type of change
- New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/snippets_cta 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 54.76 **(-0.04)** | 36.53 **(-0.07)** | 33.8 **(-0.12)** | 55.37 **(-0.02)**
 :green_circle: | app/client/src/actions/apiPaneActions.ts | 67.74 **(-0.12)** | 100 **(0)** | 0 **(0)** | 55 **(2.37)**
 :green_circle: | app/client/src/components/ads/Icon.tsx | 55.34 **(0.44)** | 37.98 **(0.48)** | 100 **(0)** | 55.19 **(0.44)**
 :sparkles: :new: | **app/client/src/components/ads/SnippetButton.tsx** | **93.75** | **83.33** | **50** | **93.75**
 :green_circle: | app/client/src/components/editorComponents/CodeEditor/generateQuickCommands.tsx | 29.87 **(0.92)** | 6.25 **(0)** | 6.25 **(0)** | 28.95 **(0.95)**
 :green_circle: | app/client/src/components/editorComponents/CodeEditor/index.tsx | 67.53 **(0.12)** | 39.75 **(0)** | 52.5 **(0)** | 68.27 **(0.12)**
 :green_circle: | app/client/src/components/editorComponents/Debugger/ContextualMenu.tsx | 45.24 **(0.94)** | 24.39 **(4.78)** | 9.52 **(-1.01)** | 45.78 **(1.48)**
 :green_circle: | app/client/src/components/editorComponents/Debugger/LogItem.tsx | 36.36 **(0)** | 23.81 **(2.07)** | 0 **(0)** | 37.1 **(0)**
 :red_circle: | app/client/src/components/editorComponents/GlobalSearch/SetSearchResults.tsx | 35.71 **(-0.65)** | 0 **(-100)** | 0 **(0)** | 41.67 **(-2.77)**
 :sparkles: :new: | **app/client/src/components/editorComponents/GlobalSearch/SnippetsDescription.tsx** | **27.12** | **3.77** | **0** | **29.52**
 :red_circle: | app/client/src/components/editorComponents/GlobalSearch/index.tsx | 53 **(-0.31)** | 14.74 **(-0.92)** | 37.31 **(-0.57)** | 54.65 **(-0.04)**
 :red_circle: | app/client/src/constants/messages.ts | 74.56 **(-0.06)** | 100 **(0)** | 24.13 **(-0.18)** | 80.98 **(0.11)**
 :green_circle: | app/client/src/entities/Action/index.ts | 78.26 **(6.04)** | 60 **(10)** | 75 **(8.33)** | 85 **(5)**
 :green_circle: | app/client/src/pages/Editor/EditorHeader.tsx | 87.83 **(0.44)** | 53.19 **(0)** | 70 **(3.33)** | 87.27 **(0.35)**
 :red_circle: | app/client/src/pages/Editor/APIEditor/Form.tsx | 38.27 **(0.77)** | 42 **(-1.75)** | 0 **(0)** | 40 **(0.78)**
 :green_circle: | app/client/src/pages/Editor/PropertyPane/index.tsx | 83.62 **(0.29)** | 69.3 **(0)** | 70.37 **(0)** | 85.32 **(0.27)**
 :green_circle: | app/client/src/pages/Editor/QueryEditor/EditorJSONtoForm.tsx | 39.13 **(0.67)** | 30.97 **(-0.82)** | 0 **(0)** | 41.38 **(0.68)**
 :green_circle: | app/client/src/reducers/uiReducers/globalSearchReducer.ts | 52.63 **(8.88)** | 33.33 **(33.33)** | 18.18 **(8.18)** | 52.63 **(8.88)**
 :green_circle: | app/client/src/sagas/ActionSagas.ts | 10.99 **(0.11)** | 0.69 **(-0.09)** | 3.7 **(0)** | 13.38 **(0.24)**
 :green_circle: | app/client/src/transformers/RestActionTransformer.ts | 96.15 **(0.32)** | 82.61 **(0.79)** | 100 **(0)** | 100 **(0)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 50.71 **(0.24)** | 39.13 **(0.87)** | 36.21 **(0)** | 55.35 **(0.27)**</details>

### CODE REVIEW

File: app/client/src/pages/Editor/EditorHeader.tsx [hunk 1]
```diff
@@ -173,6 +174,11 @@ type EditorHeaderProps = {
   inOnboarding: boolean;
 };
 
+const GlobalSearch = lazy(() => {
+  debugger;
```
Comment (rishabhsaxena) [inline app/client/src/pages/Editor/EditorHeader.tsx:178]: left over `debugger` statement

File: app/client/src/components/ads/SnippetButton.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    height: 18px;
+  }
+`;
+
+export default function SearchSnippets(props: Props) {
+  const dispatch = useDispatch();
+  const className = props.className || "";
+  function handleClick() {
+    dispatch({
+      type: "EXECUTE_COMMAND",
```
Comment (rishabhsaxena) [inline app/client/src/components/ads/SnippetButton.tsx:31]: nit: should use an action creator here for the slash command action?

File: app/client/src/components/ads/SnippetButton.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      payload: {
+        actionType: "NEW_SNIPPET",
```
Comment (rishabhsaxena) [inline app/client/src/components/ads/SnippetButton.tsx:33]: nit: should we use enums here?

File: app/client/src/components/editorComponents/GlobalSearch/SetSearchResults.tsx [hunk 4]
```diff
@@ -16,7 +16,10 @@ type Props = {
 function SearchResults({ category, hits, setSearchResults }: Props) {
   const debouncedSetter = useCallback(debounce(setSearchResults, 100), []);
   useEffect(() => {
-    debouncedSetter(hits as any, category);
+    const filteredHits = hits.filter(
+      (hit) => !hit.kind || hit.kind === SEARCH_ITEM_TYPES.document,
+    );
```
Comment (rishabhsaxena) [inline app/client/src/components/editorComponents/GlobalSearch/SetSearchResults.tsx:21]: nit: should we add a comment here, why a filter is needed?

File: app/client/src/components/editorComponents/GlobalSearch/SnippetsDescription.tsx [hunk 5]
```diff
@@ -373,9 +381,11 @@ export default function SnippetDescription({ item }: { item: Snippet }) {
     <SnippetContainer>
       <div className="snippet-title">
         <span>{title}</span>
-        {shouldInsertSnippet && (
+        {selectedIndex === 0 && (
```
Comment (rishabhsaxena) [inline app/client/src/components/editorComponents/GlobalSearch/SnippetsDescription.tsx:384]: QQ: why do need to check for `selectedIndex === 0` here?

File: app/client/src/reducers/uiReducers/globalSearchReducer.ts [hunk 6]
```diff
@@ -34,11 +39,13 @@ const globalSearchReducer = createReducer(initialState, {
...
-      ...state.filterContext,
-      category: action.payload,
-      insertSnippet: false,
-    },
+    filterContext: state.modalOpen
+      ? initialState.filterContext
+      : {
+          ...state.filterContext,
+          category: action.payload,
+          onEnter: SnippetAction.COPY,
```
Comment (rishabhsaxena) [inline app/client/src/reducers/uiReducers/globalSearchReducer.ts:47]: QQ: won't this make the action as `SnippetAction.COPY` each time the modal opens?

**Review by rishabhsaxena — DISMISSED** [review]
few nitpicks, LGTM!
also think we should consider adding a couple of basic tests for the ideal cases

File: app/client/src/reducers/uiReducers/globalSearchReducer.ts (hunk 6 above)
Comment (arunvjn) [reply app/client/src/reducers/uiReducers/globalSearchReducer.ts:47]: Yes. It should be set to INSERT only when it is triggered from a field (using slash commands)

File: app/client/src/components/editorComponents/GlobalSearch/SnippetsDescription.tsx (hunk 5 above)
Comment (arunvjn) [reply app/client/src/components/editorComponents/GlobalSearch/SnippetsDescription.tsx:384]: We need to show the message "Hit enter to insert/copy" only in snippets tab.

**Review by rishabhsaxena — APPROVED** [review]
LGTM

**Review by rishabhsaxena — APPROVED** [review]
LGTM

**Pull request merged**